xapian-bindings/java/java.i

```diff
@@ -174,8 +174,11 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next(int) "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
+    Xapian::docid next (int pseudoArg) {
```

Comment:
The exception will only be thrown if we take an iterator from MSet and use it in a while loop, forcing next() calls without checking the existence of next item with hasNext(). If we use the MSet in the for each loop we won't face any problem (next() can't even be called since there is no access to the iterator object) and no exceptions will be thrown.

What would you suggest? I could add a hasNext() check in the next() function.